src/app/icd/client/CheckInHandler.cpp

```diff
@@ -127,6 +128,9 @@ CHIP_ERROR CheckInHandler::OnMessageReceived(Messaging::ExchangeContext * ec, co
     else
     {
         mpCheckInDelegate->OnCheckInComplete(clientInfo);
+#if CHIP_CONFIG_ENABLE_READ_CLIENT
+        InteractionModelEngine::GetInstance()->OnActiveModeNotification(clientInfo.peer_node);
```

Comment:
I bet people will get upset about the singleton use here and the fact that the InteractionModelEngine is not injected...

Reply:
updated, thanks